js_modules/dagster-ui/packages/ui-core/src/instance/InstanceConcurrency.tsx

```diff
@@ -512,91 +604,111 @@ const DeleteConcurrencyLimitDialog = ({
   );
 };
 
-const ConcurrencyRunsDialog = ({
-  runIds,
+const ConcurrencyActionMenu = ({
+  pendingStep,
+  onUpdate,
+}: {
+  pendingStep: ConcurrencyStepFragment;
+  onUpdate: () => void;
+}) => {
+  const [freeSlots] = useMutation<
+    FreeConcurrencySlotsMutation,
+    FreeConcurrencySlotsMutationVariables
+  >(FREE_CONCURRENCY_SLOTS_MUTATION);
+
+  return (
+    <Popover
+      content={
+        <Menu>
+          <MenuItem
+            key="free-concurrency-slots-step"
+            icon="status"
+            text="Free concurrency slot for step"
+            onClick={async () => {
+              const resp = await freeSlots({
+                variables: {runId: pendingStep.runId, stepKey: pendingStep.stepKey},
+              });
+              if (resp.data?.freeConcurrencySlots) {
+                onUpdate();
+                await showSharedToaster({
+                  intent: 'success',
+                  icon: 'copy_to_clipboard_done',
+                  message: 'Freed concurrency slot',
+                });
+              }
+            }}
+          />
+          <MenuItem
+            key="free-concurrency-slots-run"
+            icon="status"
+            text="Free all concurrency slots for run"
+            onClick={async () => {
+              const resp = await freeSlots({variables: {runId: pendingStep.runId}});
+              if (resp.data?.freeConcurrencySlots) {
+                onUpdate();
+                await showSharedToaster({
+                  intent: 'success',
+                  icon: 'copy_to_clipboard_done',
+                  message: 'Freed concurrency slots',
+                });
+              }
+            }}
+          />
+        </Menu>
+      }
+      position="bottom-right"
+    >
+      <Button rightIcon={<Icon name="expand_more" />}>Actions</Button>
```

Comment:
Does this button have `intent="primary"` in the video? I think it should be a transparent button instead of a black one, so I think it should not have an intent applied.